src/main/java/com/thealgorithms/maths/MatrixRank.java

```diff
@@ -0,0 +1,62 @@
+package com.thealgorithms.maths;
+
+/**
+ *
+ *
+ * <h1>Rank of Matrix</h1>
+ *
+ * <p>
+ * The rank of a matrix is the largest number of linearly independent rows/columns of the matrix.
+ * The rank is not only defined for square matrices.
+ * </p>
+ *
+ * <p>
+ * The rank of a matrix can also be defined as the largest order of any non-zero minor in the matrix.
+ * </p>
+ *
+ * <p> Read more about rank of matrix <a href="https://en.wikipedia.org/wiki/Rank_(linear_algebra)">here</a></p>
+ *
+ * @author Anup Omkar
+ */
+public class MatrixRank {
+
+    // Small value to handle precision issues with floating point numbers
+    private final static double EPSILON = 1e-10;
```

Comment:
I would suggest to introduce a method of the type: `bool isZero(final double x)`.

In general I am not sure how to handle this properly. If you will have some suggestions please let me know.

Reply:
I think the simplest way would be to add a new argument `double epsilon = 10e-10` to `computeRank` method. So also the method `isZero` should take an additional argument.

Reply:
Also why do we need this isZero() method, isn't it overkilling for just one check?

Reply:
You are right. For time being, just add a `isZero` method with hard coded `epsilon`.

Reply:
Done. Please review the lastet changes.